src/variational/VariationalInference.jl

```diff
@@ -76,17 +61,29 @@ function grad!(
 end
 
 """
-    vi(model::Model, alg::VariationalInference)
-    vi(model::Model, alg::VariationalInference, q::VariationalPosterior)
+    vi(model, alg::VariationalInference)
+    vi(model, alg::VariationalInference, q::VariationalPosterior)
+    vi(model, alg::VariationalInference, getq::Function, θ::AbstractArray)
 
 Constructs the variational posterior from the `model` and performs the optimization
 following the configuration of the given `VariationalInference` instance.
+
+# Arguments
+- `model`: `Turing.Model` or `Function` z ↦ log p(x, z) where `x` denotes the observations
+- `alg`: the VI algorithm used
+- `q`: a `VariationalPosterior` for which it is assumed a specialized implementation of the variational objective used exists.
+- `getq`: function taking parameters `θ` as input and returns a `VariationalPosterior`
+- `θ`: only required if `getq` is used, in which case it is the initial parameters for the variational posterior
 """
-function vi(model::Model, alg::VariationalInference)
+function vi(model, alg::VariationalInference)
```

Comment:
Maybe replace the function with `vi(model, alg:: VariationalInference ) = throw(MethodError(vi, model, alg))`; same for the other two that comes next

Reply:
Can't you remove the lines completely if you just want to throw a `MethodError`? If there exists no suitable method, a method error will be thrown anyways. If some "empty" definition of `vi` is required, you could just write
```julia
function vi end
```
Then users/developers can implement `vi` but you don't have to provide a default implementation (and if there is no suitable method, a `MethodError` will be thrown).

Reply:
Ah, that's true!
I'll do this for `vi`, `grad!` and `optimize! then:)